hooks/tk-multi-reviewsubmission/preprocess_nuke.py

```diff
@@ -30,16 +30,27 @@ def get_processed_script(self, nuke_script_path, **kwargs):
                                                       fields=sg_fields))
         if context.task:
             sg_fields = self.parent.get_setting('task_burnin_sg_fields')
-            tasks = self.parent.shotgun.find_one('Task', filters=[['entity', 'is', context.entity],
-                                                                  ['id', 'is', context.task['id']]],
-                                                 fields=sg_fields)
-            replace_data.update(tasks)
-            if tasks.get('duration'):
-                duration = str(tasks['duration']/(8.0*60.0))+' day(s)' if tasks['duration'] else 'None'
-                replace_data.update({'duration': duration})
-            if tasks.get('time_logs_sum'):
-                time_logs_sum = str(tasks['time_logs_sum']/(8.0*60.0))+' day(s)' if tasks['time_logs_sum'] else 'None'
-                replace_data.update({'time_logs_sum': time_logs_sum})
+            if 'duration' in sg_fields and 'time_logs_sum' in sg_fields:
+                sg_data = self.parent.shotgun.find('Task', filters=[['entity', 'is', context.entity],
+                                                                    ['step', 'is', context.step]],
+                                                   fields=['duration', 'time_logs_sum'])
+                total_duration = 0
+                total_time_logged = 0
+                for data in sg_data:
+                    total_duration += data['duration'] if data['duration'] else 0
+                    total_time_logged += data['time_logs_sum'] if data['time_logs_sum'] else 0
+                if total_duration: total_duration = str(total_duration/(8.0 * 60.0)) + ' day(s)'
+                if total_time_logged: total_time_logged = str(total_time_logged/(8.0 * 60.0)) + ' day(s)'
```

Comment:
Will it always be 8hrs per day, this logic should be left to client side to handle I think... we should just pass the total value instead of giving the days calculation. And days calculation could be done in the nuke script... since every show could have different metrics.

Reply:
This cannot be show based.
8hrs per day is based on what is set in shotgun http://shotgun.d2-india.com/page/1076 under Scheduling. I'm not sure if this will ever be changed but if you want we can query it from here instead of hard-coding in the script.
All time base fields in shotgun are converted to days based on this.

Reply:
Ok then approving.